bin/convert-mobile.js

```diff
@@ -42,118 +48,171 @@ function convertSingleIconToPng(iconName, width, height, outputRoot, outputSuffi
 
 /**
  * Creates an ImageSet (including Contents.json file) for an icon
- * @param {string} iconName - name of the icon without size or extension (e.g. '2d-explore')
- * @param {int} baseSize - base (smallest) size of the icon at @1x
- * @param {string} outputRoot - base directory in which to store the imageset folder
+ * @param {string} svgFilePath - filepath to icon .svg
+ * @param {int} width - width of output file @1x
+ * @param {int} height - height of output file @1x
+ * @param {string} outputBasePath - base directory in which to store output
+ * @param {string} outputName - name of output png image, excluding '.png'
  */
-function convertIconToXcodeImageSet(iconName, baseSize, outputRoot) {
-  const new_output_root = path.join(outputRoot, iconName + '.imageset');
+function convertIconToXcodeImageSet(svgFilePath, width, height, outputBasePath, outputName) {
+  const outputImagesetPath = path.join(outputBasePath, outputName + '.imageset');
 
   // Create images at 3 sizes
-  convertSingleIconToPng(iconName, baseSize, baseSize, new_output_root, '@1x.png');
-  convertSingleIconToPng(iconName, baseSize * 2, baseSize * 2, new_output_root, '@2x.png');
-  convertSingleIconToPng(iconName, baseSize * 3, baseSize * 3, new_output_root, '@3x.png');
+  convertSingleIconToPng(svgFilePath, width, height, outputImagesetPath, outputName, '@1x');
+  convertSingleIconToPng(svgFilePath, width * 2, height * 2, outputImagesetPath, outputName, '@2x');
+  convertSingleIconToPng(svgFilePath, width * 3, height * 3, outputImagesetPath, outputName, '@3x');
 
   // read template
-  const template_path = path.join(__dirname, 'templates', 'imageset.json');
-
-  fs.readFile(template_path, 'utf8', function (error, buffer) {
-    if (error){
+  const imagesetTemplatePath = path.join(__dirname, 'templates', 'imageset.json');
+  // create Contents.json for asset catalog asset
+  fs.readFile(imagesetTemplatePath, 'utf8', function (error, buffer) {
```

Comment:
There is now a [promises-based version](https://nodejs.org/api/fs.html#fs_promises_api) of the `fs` API. I see you're using `async/await` elsewhere so as long as we don't care about Node version support we could just use that and be cool kids. Basically you change the top level import from `require("fs")` to `require("fs/promises")`. Something to think about.